io/json/include/detray/io/json_geometry_io.hpp

```diff
@@ -0,0 +1,184 @@
+/** Detray library, part of the ACTS project (R&D line)
+ *
+ * (c) 2022 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#pragma once
+
+#include <array>
+
+#include "detray/io/json_algebra_io.hpp"
+#include "detray/io/json_defs.hpp"
+#include "detray/io/json_material_io.hpp"
+#include "detray/io/json_utilities_io.hpp"
+
+/// @brief  The detray JSON I/O is written in such a way that it
+/// can read/write ACTS files that are written with the Detray
+/// JSON I/O extension
+namespace detray {
+
+/// @brief  A payload object for masks
+struct mask_payload {
+    enum class mask_type : unsigned int {
+        annulus2 = 0u,
+        cuboid3 = 1u,
+        cylinder2 = 2u,
+        cylinder3 = 3u,
+        line = 4u,
+        rectangle2 = 5u,
+        ring2 = 6u,
+        single3 = 7u,
+        trapezoid2 = 8u
+    };
+    mask_type type = mask_type::ring2;
+    std::vector<real_io> boundaries;
+};
+
+void to_json(nlohmann::json& j, const mask_payload& m) {
+    j["type"] = static_cast<unsigned int>(m.type);
+    j["boundaries"] = m.boundaries;
+}
+
+void from_json(const nlohmann::json& j, mask_payload& m) {
+    m.type = static_cast<mask_payload::mask_type>(j["type"]);
+    m.boundaries = j["boundaries"].get<std::vector<real_io>>();
+}
+
+/// @brief  A payload for surfaces
+struct surface_payload {
+    transform_payload transform;
+    mask_payload mask;
+    material_slab_payload material;
+    std::size_t gid;
```

Comment:
Hm, this is more difficult: That is already the surface index relative to the volume or still an ACTS geoID? A ```geo_obj_id``` in detray tells a volume if it must e.g. distinguish between modules and portals, so maybe we can call this an index?

Reply:
The idea to add the `gid` is to associate event data, e.g. from the `csv` writer uniquely.